### PR TITLE
Adding image upload views for admins and profiles.

### PR DESCRIPTION
**Pull request by dessalines**
## Description

Adds the ability to view and delete media uploads for admins and users

## Screenshots

https://github.com/LemmyNet/lemmy-ui/assets/930215/d2c942ea-e582-4aa4-87e8-fb00cd1e6d5a

context: https://github.com/LemmyNet/lemmy/pull/4631

### CODE REVIEW

File: src/shared/components/common/media-uploads.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      <button
+        onClick={linkEvent({ i: this, image }, this.handleDeleteImage)}
+        className="btn btn-danger"
+      >
+        {I18NextService.i18n.t("delete")}
+      </button>
+    );
+  }
+
+  async handleDeleteImage({ image }: { i: MediaUploads; image: LocalImage }) {
```
Comment (SleeplessOne1917) [inline src/shared/components/common/media-uploads.tsx:80]: You're not currently using `i`. You don't need it.

File: src/shared/components/common/media-uploads.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+
+  deleteImageBtn(image: LocalImage) {
+    return (
+      <button
+        onClick={linkEvent({ i: this, image }, this.handleDeleteImage)}
```
Comment (SleeplessOne1917) [inline src/shared/components/common/media-uploads.tsx:72]: The value for `i` never gets used in `handleDeleteImage`. IDK if `linkEvent` still has optimization benefits if you're not trying to use `this`.

Comment (dessalines) [reply src/shared/components/common/media-uploads.tsx:72]: I'll try keeping linkEvent, but removing this.

**Comment by dessalines** [comment]
The blank pages are gonna be tough to figure out, bc I can't replicate. I just tested again, and the linked video above shows a refresh. It might be because this requires an unmerged back-end change that's running with my new `LocalImageView` . I'll leave this as draft until the back-end gets merged.

**Review by SleeplessOne1917 — APPROVED** [review]
Tested it again and it works.

**Pull request merged**